src/backend/cuda/math.hpp

```diff
@@ -54,6 +54,10 @@ static inline __DH__ size_t max(size_t lhs, size_t rhs) {
 }
 
 #ifdef __CUDA_ARCH__
+static inline __device__ __half abs(__half val) {
+    return __short_as_half(__half_as_short(val) & 0x7FFF);
```

Comment:
It's unfortunate that they have lot half `h*` math functions but not `habs`.

Reply:
Yeah, I doubt this is an expensive operation. Those are reinterpret casts so basically noops.